File: backend/app/schemas/import_mapping.py
"""
Pydantic schemas for Import Mapping API endpoints.
"""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum


class MappingType(str, Enum):
    """Type of import mapping."""

    PLATFORM = "platform"
    STOREFRONT = "storefront"


# ============================================================================
# Response Schemas
# ============================================================================


class ImportMappingResponse(BaseModel):
    """Response schema for a single import mapping."""

    model_config = {"from_attributes": True}

    id: str
    user_id: str
    import_source: str
    mapping_type: MappingType
    source_value: str
    target_id: str
    created_at: datetime
    updated_at: datetime


class ImportMappingListResponse(BaseModel):
    """Response schema for listing import mappings."""

    items: List[ImportMappingResponse]
    total: int


# ============================================================================
# Request Schemas
# ============================================================================


class ImportMappingCreateRequest(BaseModel):
    """Request schema for creating an import mapping."""

    import_source: str = Field(..., min_length=1, max_length=50)
    mapping_type: MappingType
    source_value: str = Field(..., min_length=1, max_length=255)
    target_id: str = Field(..., min_length=1, max_length=100)


class ImportMappingUpdateRequest(BaseModel):
    """Request schema for updating an import mapping."""

    target_id: str = Field(..., min_length=1, max_length=100)


class BatchMappingItem(BaseModel):
    """A single mapping in a batch request."""

    mapping_type: MappingType
    source_value: str = Field(..., min_length=1, max_length=255)
    target_id: str = Field(..., min_length=1, max_length=100)


class BatchImportMappingRequest(BaseModel):
    """Request schema for batch creating/updating import mappings."""

    import_source: str = Field(..., min_length=1, max_length=50)
    mappings: List[BatchMappingItem]


class BatchImportMappingResponse(BaseModel):
    """Response schema for batch import mapping operation."""

    created: int
    updated: int

File: frontend/src/types/import-mapping.ts
/**
 * Types for import mapping functionality.
 *
 * Import mappings store user preferences for how import source values
 * (e.g., "PC", "Steam") should map to platform/storefront IDs.
 */

export enum MappingType {
  PLATFORM = 'platform',
  STOREFRONT = 'storefront',
}

export interface ImportMapping {
  id: string;
  userId: string;
  importSource: string;
  mappingType: MappingType;
  sourceValue: string;
  targetId: string;
  createdAt: string;
  updatedAt: string;
}

export interface ImportMappingListResponse {
  items: ImportMapping[];
  total: number;
}

export interface CreateImportMappingRequest {
  importSource: string;
  mappingType: MappingType;
  sourceValue: string;
  targetId: string;
}

export interface BatchMappingItem {
  mappingType: MappingType;
  sourceValue: string;
  targetId: string;
}

export interface BatchImportMappingResponse {
  created: number;
  updated: number;
}
